runners: jlink: Add support for J-Link ethernet

marnold-b opened the pull request:
Add support for J-Link over IP and J-Link remote server.
If the "--dev-id" is a valid ip, the transport over ip is selected.
Otherwise usb is selected.

Command to flash:
```
west -v flash --dev-id=<ip> -r jlink
```

carlescufi commented on scripts/west_commands/runners/jlink.py:
```diff
@@ -35,6 +35,7 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
 
     def __init__(self, cfg, device, dev_id=None,
                  commander=DEFAULT_JLINK_EXE,
+                 select="usb",
```
I think we should name this `transport` instead, because we may want to generalize this in the future, given that several debuggers support this.

carlescufi commented:
Also see https://github.com/zephyrproject-rtos/zephyr/pull/52140.

carlescufi commented on scripts/west_commands/runners/jlink.py:
```diff
@@ -80,7 +82,9 @@ def capabilities(cls):
     def dev_id_help(cls) -> str:
         return '''Device identifier. Use it to select the J-Link Serial Number
-                  of the device connected over USB.'''
+                  of the device connected over USB. If the J-Link over ethernet
```
Please do not use "ethernet" anywhere. Instead please use "ip" or "IP".

marnold-b commented:
From my side the requested changes are addressed.
I am also open to the suggested solution in https://github.com/zephyrproject-rtos/zephyr/pull/64915#discussion_r1384923011, just let me know which solution is preferred.

jukkar commented on scripts/west_commands/runners/jlink.py:
```diff
@@ -96,6 +100,8 @@ def do_add_parser(cls, parser):
                             help='specifies a loader type')
         parser.add_argument('--id', required=False, dest='dev_id',
                             help='obsolete synonym for -i/--dev-id')
+        parser.add_argument('--select', default='usb',
```
Using dev_id for ip address sounds good. +1 from me.

marnold-b replied:
@carlescufi I updated the PR to your suggested solution.

marnold-b committed:
runners: jlink: Add support for J-Link over IP

Add support for J-Link over IP and J-Link remote server.
If the "--dev-id" is a valid ip, the transport over ip is selected.
Otherwise usb is selected.

Signed-off-by: Michael Arnold <marnold@baumer.com>

marnold-b commented:
@carlescufi can you have another look?
